substrafl_examples/get_started/run_diabetes_substrafl.py

```diff
@@ -0,0 +1,573 @@
+            _, next_shared_state = node.update_states(
+                self.local_second_order_computation(
+                    node.data_sample_keys,
+                    shared_state=self.first_order_aggregated_state,
+                    _algo_name=f"Computing second order analytics with {self.__class__.__name__}",
+                ),
+                local_state=local_states[node.organization_id],
+                round_idx=1,
+                authorized_ids=set([node.organization_id]),
+                aggregation_id=aggregation_node.organization_id,
+                clean_models=False,
+            )
+
+            # All local analytics are stored in the second_order_shared_states,
+            # given as input the the aggregation method.
+            second_order_shared_states.append(next_shared_state)
+
+        # Call the aggregation method on the second_order_shared_states
+        self.second_order_aggregated_state = aggregation_node.update_states(
+            self.aggregation(
+                shared_states=second_order_shared_states,
+                _algo_name="Aggregating second order",
+            ),
+            round_idx=1,
+            authorized_ids=set(
+                [
+                    train_data_node.organization_id
+                    for train_data_node in train_data_nodes
+                ]
+            ),
+            clean_models=False,
+        )
+
+    def save_local_state(self, path: pathlib.Path):
+        """This function will save the important local state to retrieve after each new
+        call to a train or test task.
+
+        Args:
+            path (pathlib.Path): Path where to save the local_state. Provided internally by
+                Substra.
+        """
+        state_to_save = {
+            "first_order": self.first_order_aggregated_state,
+            "second_order": self.second_order_aggregated_state,
+        }
+        with open(path, "w") as f:
+            json.dump(state_to_save, f)
+
+    def load_local_state(self, path: pathlib.Path):
+        """Mirror function to load the local_state from a file saved using
+        ``save_local_state``.
+
+        Args:
+            path (pathlib.Path): Path where to load the local_state. Provided internally by
+                Substra.
+
+        Returns:
+            ComputePlanBuilder: return self with the updated local state.
+        """
+        with open(path, "r") as f:
+            state_to_load = json.load(f)
+
+        self.first_order_aggregated_state = state_to_load["first_order"]
+        self.second_order_aggregated_state = state_to_load["second_order"]
+
+        return self
+
+
+# %%
+# Running the experiment
+# ======================
+#
+# As a last step before launching our experiment, we need to specify the third parties dependencies required to run it.
+# The :ref:`substrafl_doc/api/dependency:Dependency` object is instantiated in order to install the right libraries in
+# the Python environment of each organization.
```

Comment:
Shall the Dependency object definition be moved to its own cell here?

Reply:
If we move the example to "go further", I would say no